src/PostgREST/QueryBuilder.hs

```diff
@@ -321,19 +327,19 @@ unquoted (JSON.Number n) =
 unquoted (JSON.Bool b) = show b
 unquoted v = toS $ JSON.encode v
 
--- private functions
 asCsvF :: SqlFragment
 asCsvF = asCsvHeaderF <> " || '\n' || " <> asCsvBodyF
   where
-    asCsvHeaderF =
-      "(SELECT coalesce(string_agg(a.k, ','), '')" <>
-      "  FROM (" <>
-      "    SELECT json_object_keys(r)::TEXT as k" <>
-      "    FROM ( " <>
-      "      SELECT row_to_json(hh) as r from " <> sourceCTEName <> " as hh limit 1" <>
-      "    ) s" <>
-      "  ) a" <>
-      ")"
```

Comment:
i was actually thinking of removing `qc` from the code since it's very inefficient when it comes to "big payloads".
We've tracked this down to being the culprit ob big memory usage.
I would say leave this part as it is and in another PR we'll fix the other sections
(there is a issue somehwere with a minimal test case that shows the memory consumption issue)

Reply:
In this case there's no input interpolation, no big payload chance there, in the out of memory rpc issue the args to the proc are being interpolated.

Reply:
the idea is if this lib does not work well in all the cases maybe it's better to not include it at all, it only adds a bit of "nice strings", nothing more.
maybe by not using it we drop other dependencies too (though not many i think)

Reply:
The qc library does make the string building easier to read for those queries that aren't in danger of becoming too long with the payload.  I can see the benefit of removing the library so that we never accidentally use it in the wrong place, but if we're careful it may not cause any harm. Maybe there are more efficient text building libraries to investigate too?